json-schema/schema.json

```diff
@@ -298,6 +316,52 @@
         }
       }
     },
+    "variable": {
+      "title": "Variable Object",
+      "type": "object",
+      "required": [
+        "dimensions"
+      ],
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/variable_type"
+        },
+        "description": {
+          "$ref": "#/definitions/description"
+        },
+        "dimensions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "values": {
+          "type": "array",
+          "minItems": 1
+        },
+        "extent": {
+          "type": "array",
+          "minItems": 2,
+          "maxItems": 2,
+          "items": {
+            "type": [
+              "string",
+              "number",
+              "null"
+            ]
+          }
+        },
+        "step": {
+          "type": [
+            "string",
```

Comment:
String is not allowed according to the README, instead number is. Not sure whether the schema or the README is correct...

Reply:
Good catch. I think the readme should be updated. NetCDF allows for text-based dimensions, geographic regions for example: http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#geographic-regions.

Reply:
Oh hmm, according to http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#terminology, a coordinate variable "is defined as a numeric data type with values that are ordered monotonically." But even that example gives `time` as an example. Perhaps that definition is mixing together the logical type (like "time") with the storage type (which may need to be an integer, like days since 1980-01-01).

Reply:
Thanks. @m-mohr do you know if something like this has come up elsewhere in STAC, where the storage type differs from the logical type? I don't know which would be more appropriate for inclusion in a catalog. But my inclination is that the catalog should reflect the data as is, rather than trying to implicitly layer interpretations on top. Maybe the solution is to formalize the `integer values + units='days since 1960-01-01 calendar='Proleptic Gregorian"` concept in a `cf` STAC extension.

Apologies if this is incoherent. Still thinking this through.

Reply:
What we usually do in STAC is to put a common abstraction on top of the internal structures. The goal is that clients can read things easily independant of the underlying file formats etc. For example, if there's a data cube format that stores all timestamps as unix timestamps (seconds since 1970), we would still use RFC3339 (ISO8601) timestamps in the metadata as that's the common denominator on the web. STAC is an abstraction on top of the "native" metadata for easy consumption and to have a common language. If there's a need for the native encoding then clients should fall back to the CF metadata (or a specific format specific extension, e.g. a cf STAC extension).

Reply:
OK, that seems reasonable, I think. It feels similar to the `decode_times` parameter in `xarray.from_zarrs`, `True` by default, which converts the integers to datetimes.